addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/AddonUpdateCommand.java

```diff
@@ -0,0 +1,181 @@
+
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.manager.impl.utils.CoordinateUtils;
+import org.jboss.forge.addon.projects.Project;
+import org.jboss.forge.addon.projects.ProjectFactory;
+import org.jboss.forge.addon.projects.Projects;
+import org.jboss.forge.addon.projects.facets.MetadataFacet;
+import org.jboss.forge.addon.ui.command.AbstractUICommand;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.context.UIValidationContext;
+import org.jboss.forge.addon.ui.input.InputComponent;
+import org.jboss.forge.addon.ui.input.UICompleter;
+import org.jboss.forge.addon.ui.input.UIInput;
+import org.jboss.forge.addon.ui.input.UIPrompt;
+import org.jboss.forge.addon.ui.metadata.WithAttributes;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.addon.ui.validate.UIValidator;
+import org.jboss.forge.furnace.Furnace;
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.addons.AddonId;
+import org.jboss.forge.furnace.manager.AddonManager;
+import org.jboss.forge.furnace.manager.request.InstallRequest;
+import org.jboss.forge.furnace.manager.spi.AddonDependencyResolver;
+import org.jboss.forge.furnace.manager.spi.Response;
+import org.jboss.forge.furnace.repositories.AddonRepository;
+import org.jboss.forge.furnace.versions.Versions;
+
+public class AddonUpdateCommand extends AbstractUICommand implements AddonCommandConstants
+{
+   @Inject
+   private AddonManager addonManager;
+
+   @Inject
+   private AddonDependencyResolver resolver;
+
+   @Inject
+   @WithAttributes(label = "Name", description = "The addon's \"groupId:artifactId\" name", required = true)
+   private UIInput<String> named;
+
+   @Inject
+   private ProjectFactory projectFactory;
+
+   @Inject
+   private Furnace furnace;
+
+   @Override
+   public Metadata getMetadata(UIContext context)
+   {
+      boolean gui = context.getProvider().isGUI();
+      return Metadata.from(super.getMetadata(context), getClass())
+               .name(gui ? ADDON_UPDATE_COMMAND_NAME : ADDON_UPDATE_COMMAND_NAME_NO_GUI)
+               .description(ADDON_UPDATE_COMMAND_DESCRIPTION)
+               .category(Categories.create(ADDON_MANAGER_CATEGORIES));
+   }
+
+   @Override
+   public void initializeUI(UIBuilder builder) throws Exception
+   {
+      Project project = Projects.getSelectedProject(projectFactory, builder.getUIContext());
+      if (project != null)
+      {
+         MetadataFacet facet = project.getFacet(MetadataFacet.class);
+         Coordinate c = facet.getOutputDependency().getCoordinate();
+         named.setDefaultValue(AddonId.from(c.getGroupId() + ":" + c.getArtifactId(), c.getVersion())
```

Comment:
Oops, it is the copy/paste issue.

Will write tests, however I was not sure I go the right direction with this command. However there is no tests for install neither, is there?